content/700-optimize/400-recommendations/100-excessive-number-of-rows-returned.mdx

```diff
@@ -0,0 +1,45 @@
+---
+title: 'Excessive number of rows returned'
+metaTitle: 'Optimize Recommendations: Excessive number of rows returned'
+metaDescription: "Learn about the recommendation provided by Optimize for excessive number of rows returned."
+tocDepth: 3
+toc: true
+---
+
+The following query targeting a `User` model do not provide a [`take` option](/orm/reference/prisma-client-reference#findmany).
+
+```ts
+await user.findMany({ select: { email: "niko@prisma.io", }})
+```
+
+## Why is this a problem?
+
+When you execute a query without specifying a limit, the query will return all relevant rows. This can be undesirable for several reasons:
+
+### User experience
+
+- **Viewing data:** Users typically need only a portion of the data at any given time, rather than all the data at once
+- **Impact on the user's device:** Displaying all the data at once can burden the user's device resources. For example, loading thousands of rows of data in a web application can cause the browser to become slow or unresponsive, consuming a significant amount of the user's device memory and CPU to display this data
+- **Waiting time:** Retrieving a large number of rows can significantly increase the time it takes get the data from the database to the user's own device.
```

Comment:
- **Waiting time:** Retrieving a large number of rows can significantly increase the time it takes to get the data from the database to the user's device.